src/libraries/Microsoft.PowerFx.Interpreter/Functions/Mutation/RemoveFunction.cs

```diff
@@ -158,10 +156,9 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
                         errors.EnsureError(DocumentErrorSeverity.Severe, args[i], ErrTableDoesNotAcceptThisType);
                     }
                 }
-            }
-
-            // Remove returns the new collection, so the return schema is the same as the collection schema.
-            returnType = collectionType;
+            }
+
+            returnType = context.Features.PowerFxV1CompatibilityRules ? DType.ObjNull : DType.EmptyRecord;
```

Comment:
>DType.EmptyRecor [](http://example.com/codeflow?start=88&length=16)

`DType.Boolean` - that's the legacy return type

Reply:
As discussed offline this is updated to be DType.Table